lib/ts/recipe/emailverification/recipeImplementation.ts

```diff
@@ -0,0 +1,184 @@
+    const querier = new Querier(recipeId, appInfo);
+    return {
+        verifyEmail: async function ({
+            token,
+            config,
+            options,
+        }: {
+            token?: string;
+            config: NormalisedInputType;
+            options?: RecipeFunctionOptions;
+        }): Promise<{
+            status: "EMAIL_VERIFICATION_INVALID_TOKEN_ERROR" | "OK" | "CUSTOM_RESPONSE";
+            fetchResponse?: Response;
+        }> {
+            token = token === undefined ? getQueryParams("token") : token;
+
+            if (token === undefined) {
+                return {
+                    status: "EMAIL_VERIFICATION_INVALID_TOKEN_ERROR",
+                };
+            }
+
+            const { json, fetchResponse } = await querier.post<
+                | {
+                      status: "OK" | "EMAIL_VERIFICATION_INVALID_TOKEN_ERROR";
+                  }
+                | APIGeneralError
+            >(
+                EMAIL_VERIFY_PATH,
+                {
+                    body: JSON.stringify({
+                        method: "token",
+                        token,
+                    }),
+                },
+                async (context) => {
+                    let postRecipeHookContext = await config.preAPIHook({
+                        ...context,
+                        action: "VERIFY_EMAIL",
+                    });
+
+                    if (options === undefined || options.preAPIHook === undefined) {
+                        return postRecipeHookContext;
+                    }
+
+                    return options.preAPIHook({
+                        url: postRecipeHookContext.url,
+                        requestInit: postRecipeHookContext.requestInit,
+                    });
+                }
+            );
+
+            if (json.status === "GENERAL_ERROR") {
+                throw new Error(json.message);
+            }
+
+            if (json.status !== "OK" && json.status !== "EMAIL_VERIFICATION_INVALID_TOKEN_ERROR") {
+                return {
+                    status: "CUSTOM_RESPONSE",
+                    fetchResponse,
+                };
+            }
+
+            return {
+                status: json.status,
+                fetchResponse,
+            };
+        },
+
+        isEmailVerified: async function ({
+            config,
+            options,
+        }: {
+            config: NormalisedInputType;
+            options?: RecipeFunctionOptions;
+        }): Promise<
+            | {
+                  status: "OK";
+                  isVerified: boolean;
+                  fetchResponse: Response;
+              }
+            | {
+                  status: "CUSTOM_RESPONSE";
+                  fetchResponse: Response;
+              }
+        > {
+            const { json, fetchResponse } = await querier.get<{ status: "OK"; isVerified: boolean } | APIGeneralError>(
+                EMAIL_VERIFY_PATH,
```

Comment:
I prefer hard coded paths here instead of constants

Reply:
Done